File: nba-tracker/src/utils/season.ts
// Get current NBA season (e.g., "2024-25")
// Seasons start in October, so Oct-Dec uses current year, Jan-Sep uses previous year
export const getCurrentSeason = (): string => {
  const now = new Date();
  const currentYear = now.getFullYear();
  const currentMonth = now.getMonth() + 1;

  if (currentMonth >= 10) {
    return `${currentYear}-${(currentYear + 1).toString().slice(2)}`;
  } else {
    return `${currentYear - 1}-${currentYear.toString().slice(2)}`;
  }
};

// Build list of recent seasons for dropdowns
// Returns current season plus previous ones going backwards
export const getSeasonOptions = (numSeasons: number = 5): string[] => {
  const seasons: string[] = [];
  const currentYear = new Date().getFullYear();
  const currentMonth = new Date().getMonth() + 1;
  const currentSeasonStartYear = currentMonth >= 10 ? currentYear : currentYear - 1;
  for (let i = 0; i < numSeasons; i++) {
    const year = currentSeasonStartYear - i;
    const seasonStr = `${year}-${(year + 1).toString().slice(2)}`;
    seasons.push(seasonStr);
  }
  return seasons;
};

File: nba-tracker-api/app/utils/season.py
from datetime import datetime


def get_current_season() -> str:
    """
    Returns the current NBA season in YYYY-YY format.
    NBA seasons start in October, so if the current month is October or later,
    we're in the season that started that year. Otherwise, we're in the season
    that started the previous year.
    """
    now = datetime.now()
    current_year = now.year
    current_month = now.month

    if current_month >= 10:
        return f"{current_year}-{(current_year + 1) % 100:02d}"
    else:
        return f"{current_year - 1}-{current_year % 100:02d}"
